core/build.gradle.kts

```diff
@@ -64,10 +64,9 @@ kotlin {
     val commonMain by getting {
       dependencies {
         api(libs.bundles.arrow)
-
         api(libs.bundles.ktor.client)
         api(projects.xefTokenizer)
-
+        implementation(libs.openai.client)
```

Comment:
This couples core with open ai, which we tried to avoid and worked to decouple in the last weeks.
Looking at what is delegating to https://github.com/aallam/openai-kotlin/blob/main/openai-client/src/commonMain/kotlin/com.aallam.openai.client/extension/internal/CosineSimilarity.kt#L19 it seems simple enough not to include as a dependency. We could provide similar functions locally or change ours to that implementation if it's better than what we have now.